### ISSUE TITLE
Unable to create and connect to SQLite database

### ISSUE THREAD

**Issue by nigiriits** [dbgate/dbgate#653]
Greetings!

**Describe the bug**
After creating the SQLite database (File -> New SQLite database -> OK) I see an error "/home/dbgate-docker/node_modules/better-sqlite3/build/Release/better_sqlite3.node: cannot open shared object file: No such file or directory" in the section "TABLES, VIEWS, FUNCTIONS".

I tried different versions - the latest, beta and 5.2.5. The result is the same everywhere.

The full bug report looks like this:
--------------------------
Error: /home/dbgate-docker/node_modules/better-sqlite3/build/Release/better_sqlite3.node: cannot open shared object file: No such file or directory
    at Object.Module._extensions..node (internal/modules/cjs/loader.js:1173:18)
    at Module.load (internal/modules/cjs/loader.js:979:32)
    at Function.Module._load (internal/modules/cjs/loader.js:819:12)
    at Module.require (internal/modules/cjs/loader.js:1003:19)
    at require (internal/modules/cjs/helpers.js:107:18)
    at bindings (/home/dbgate-docker/node_modules/bindings/bindings.js:112:48)
    at new Database (/home/dbgate-docker/node_modules/better-sqlite3/lib/database.js:48:64)
    at Object.connect (/home/dbgate-docker/plugins/dbgate-plugin-sqlite/dist/backend.js:9:132590)
    at e.exports (/home/dbgate-docker/bundle.js:11:86941)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)

Error JSON: {
  "code": "ERR_DLOPEN_FAILED"
}

Connection: {
  "engine": "sqlite@dbgate-plugin-sqlite",
  "databaseFile": "***",
  "singleDatabase": true,
  "defaultDatabase": "***",
  "_id": "***"
}

Platform: linux
--------------------------

The command to run is: "sudo docker run -d --name dbgate-instance -v test:/root/.dbgate -p 3000:3000 dbgate/dbgate"

**Expected behavior**
The database should appear and open.

**Screenshots**
![Снимок экрана 2023-11-05 201901](https://github.com/dbgate/dbgate/assets/77151002/f8b6f6cc-4d3a-4b90-bdc9-d1992095c542)

**Version Information (please complete the following information):**
 - OS: Raspberry Pi OS x64 bullseye
 - App Version:
 - Install source: Docker
 - Type - Web
 - Database engine: SQLite

**Comment by ProjectInfinity** [comment]
Does this happen using the ARM AppImage as well or only under Docker?

**Comment by nigiriits** [comment]
> Does this happen using the ARM AppImage as well or only under Docker?

This is the first time I've heard about this! Is this what you mean? https://appimage.github.io/DbGate/

But I don't have a GUI on Raspberry Pi...

**Comment by ProjectInfinity** [comment]
I have some good news, I was able to re-create your issue and also fix it however I need to "translate" my discovery into something that fixes it without manual intervention. I will hopefully be back with more shortly.

**Comment by nigiriits** [comment]
Oh, this is great news!

**Comment by ProjectInfinity** [comment]
Could you please try the latest beta docker build? Please make sure you completely remove any old images of dbgate and the container itself.

Then run
`docker pull dbgate/dbgate:beta`
and then
`docker run -it --name dbgate-instance -p 3100:3000 dbgate/dbgate:beta`
once this is done open your browser at http://localhost:3100 and create a SQLite database.

Let me know if this fixes it for you.

**Comment by nigiriits** [comment]
@ProjectInfinity, alas, no, exactly the same error.

`nikolay@servos:~ $ sudo docker pull dbgate/dbgate:beta
beta: Pulling from dbgate/dbgate
bfbe77e41a78: Pull complete
818220eb621a: Pull complete
eadb372b7e80: Pull complete
c19f61179695: Pull complete
929469a7a079: Pull complete
2aa7343c7745: Pull complete
4f4fb700ef54: Pull complete
Digest: sha256:b73b5bd6e5a747fdab057c080beacf40e182b664cf00e23aa8c5aed1e8e80b44
Status: Downloaded newer image for dbgate/dbgate:beta
docker.io/dbgate/dbgate:beta
nikolay@servos:~ $ sudo docker run -it --name dbgate-instance -p 3100:3000 dbgate/dbgate:beta`

`{"pid":154,"caller":"requirePlugin","name":"databaseConnection","level":30,"msg":"Loading module dbgate-plugin-sqlite from /home/dbgate-docker/plugins/dbgate-plugin-sqlite/dist/backend.js","time":1699495862618}
{"pid":1,"caller":"databaseConnections","name":"main","level":50,"msg":"Error in database connection 17350b80-7ea5-11ee-af1f-4d2869c81997, database newdb222.sqlite: /home/dbgate-docker/node_modules/better-sqlite3/build/Release/better_sqlite3.node: cannot open shared object file: No such file or directory","time":1699495862782}
{"pid":154,"caller":"dbconnProcess","name":"databaseConnection","err":{"code":"ERR_DLOPEN_FAILED"},"level":50,"msg":"Error in DB connection","time":1699495862782}`

**Comment by ProjectInfinity** [comment]
Hmm, that makes me wonder if the issue is perhaps a different one on the ARM image. Do you happen to have a x86 computer with docker nearby that you can test the latest beta with just to ensure that it does in fact work out of the box?

**Comment by nigiriits** [comment]
I tried it on another Raspberry Pi OS system with `docker-ce/bullseye,now 5:24.0.7-1~debian.11~bullseye armhf`. There's another error.

->
`Error: /home/dbgate-docker/node_modules/better-sqlite3/build/Release/better_sqlite3.node: wrong ELF class: ELFCLASS64
    at Object.Module._extensions..node (internal/modules/cjs/loader.js:1173:18)
    at Module.load (internal/modules/cjs/loader.js:979:32)
    at Function.Module._load (internal/modules/cjs/loader.js:819:12)
    at Module.require (internal/modules/cjs/loader.js:1003:19)
    at require (internal/modules/cjs/helpers.js:107:18)
    at bindings (/home/dbgate-docker/node_modules/bindings/bindings.js:112:48)
    at new Database (/home/dbgate-docker/node_modules/better-sqlite3/lib/database.js:48:64)
    at Object.connect (/home/dbgate-docker/plugins/dbgate-plugin-sqlite/dist/backend.js:9:132590)
    at e.exports (/home/dbgate-docker/bundle.js:11:86941)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)

Error JSON: {
  "code": "ERR_DLOPEN_FAILED"
}

Connection: {
  "engine": "sqlite@dbgate-plugin-sqlite",
  "databaseFile": "",
  "singleDatabase": true,
  "defaultDatabase": "",
  "_id": ""
}

Platform: linux
<-

I tried the above on the system `docker-ce/bullseye,now 5:24.0.7-1~debian.11~bullseye arm64`.

There is an error here and there, but different.

_p.s.
But in Docker Windows everything works fine! Just checked._

**Comment by ProjectInfinity** [comment]
Thanks for testing. That's what I was afraid of, it seems like the issue is related to the ARM image itself. Need to dig out my RPi4 from wherever it is hiding it seems.

**Comment by nigiriits** [comment]
I'm sad to hear this. :)

**Comment by n-traore** [comment]
Hi, just to add some info.
I have been experiencing the same issue for a few weeks now on MacOS Ventura (version 5.2.6, and updated to 5.2.7 today thinking it was solved). 
I saw on [this issue](https://github.com/dbgate/dbgate/issues/614) that version 5.2.5 was OK for docker : just tested with the MacOS 5.2.5 release and I can confirm it works to open my sqlite database !

Hope it can help solve this..

**Comment by nigiriits** [comment]
Versions 5.2.5 - 5.2.7 do not work, checked! Version 5.2.8-beta.2 does not start at all, screenshot below. ;(

![Снимок экрана 2023-11-19 223347](https://github.com/dbgate/dbgate/assets/77151002/84fb86b4-7653-47f0-8c94-33bc7a06c4f3)

**Comment by nigiriits** [comment]
Hello! Is it even possible to solve this problem?

**Comment by meglio** [comment]
> Yes but it is not a simple issue to resolve as it involves multiple architectures and without a ARM device to test on it is hard to debug or verify changes.

Can I help you with debugging? Or is there a possibility to rent a macos and debug over there?

**Comment by meglio** [comment]
A temporary workaround is force AMD image and let MacOS use Rosetta2, see here:

https://github.com/dbgate/dbgate/issues/710#issuecomment-1954237994

**Comment by meglio** [comment]
Not fixed since then?

**Comment by 9a3eedi** [comment]
Hello,

I would like to mention that I am also facing this issue on an x86_64 Linux computer (Kubuntu 24.04) using the latest version (6.5.4) .deb package, so I don't think this is limited to ARM Macs. 

I was initially using the snap package, but I faced [another issue there](https://github.com/dbgate/dbgate/issues/804) and the workaround for that was to use the deb package, so I don't know if the snap version has this issue. I think I also tried the AppImage and still had the same issue, but I cannot remember.

My current workaround is to downgrade to 5.5.6 (the latest 5.x release). I don't have either of the issues with this version.
